pkg/hubble/metrics/flows-to-world/handler.go

```diff
@@ -85,6 +98,12 @@ func (h *flowsToWorldHandler) ProcessFlow(_ context.Context, flow *flowpb.Flow)
 	if flow.GetVerdict() != flowpb.Verdict_DROPPED && isReply {
 		return nil
 	}
+
+	// if "syn-only" option is set, only count non-reply SYN packets for TCP.
+	if h.synOnly && l4.GetTCP() != nil && l4.GetTCP().GetFlags() != nil && (!l4.GetTCP().GetFlags().SYN || isReply) {
```

Comment:
You can avoid the `nil` checks and just use `l4.GetTCP().GetFlags().GetSYN()`

Reply:
magical 💯